### DISCUSSION TITLE
Disable For orgMode

### DISCUSSION THREAD

**Discussion by centuryx476** [saghen/blink.cmp#575] [Q&A] [answered]
### Feature Description

Hello,
I see in #491 I can add a function to disable autocomplete for specific file types.
But how do I add it to the config?

This is all I have for the config. But if I add it inside the opts {} then autoComplete is disabled for everything. I need autoComplete to be turned off for .org and .txt files.
![image](https://github.com/user-attachments/assets/64d6dbb7-7d6e-42be-8c5d-51b364fb7435)

**Comment by Shougo** [comment]
What are the filetype?

```lua
enabled = function()
  return not vim.tbl_contains({ "lua", "markdown" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt"
end
```

You can change the filetype list.

**Comment by centuryx476** [comment]
> What are the filetype?
> 
> ```lua
> enabled = function()
>   return not vim.tbl_contains({ "lua", "markdown" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt"
> end
> ```
> 
> You can change the filetype list.

I can place the function in for example "init.lua" ?
The file types are org, txt

**Reply by centuryx476** [reply]
I wrote it into a function and then call the function from init.lua but autoComplete is still active for org files...

![image](https://github.com/user-attachments/assets/15353222-de0e-4029-8edf-eba3e32c6de8)

**Reply by vicantwin** [reply, marked as answer]
You need to put the function inside `opts` for blink's PluginSpec
You don't need to make a new function; just replace the filetypes in the function given by @Shougo.
```lua
{
    "saghen/blink.cmp",
    lazy = false, -- lazy loading handled internally
    -- optional: provides snippets for the snippet source
    dependencies = "rafamadriz/friendly-snippets",

    -- use a release tag to download pre-built binaries
    version = "v0.*",
    -- OR build from source, requires nightly: https://rust-lang.github.io/rustup/concepts/channels.html#working-with-nightly-rust
    -- build = 'cargo build --release',
    -- If you use nix, you can build from source using latest nightly rust with:
    -- build = 'nix run .#build-plugin',

    opts = {
      -- the function should be here
      enabled = function()
        return not vim.tbl_contains({ "org", "txt" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt"
      end,
    },
    -- allows extending the providers array elsewhere in your config
    -- without having to redefine it
    opts_extend = { "sources.default" },
  }
```

**Comment by esn89** [comment]
I can't seem to get the function of:

```lua
    opts = {
      -- the function should be here
      enabled = function()
        return not vim.tbl_contains({ "org", "txt" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt"
      end,
    },
 ```
 
 to work.  What happens is that, in orgmode, it still tries to use the cmdline autocomplete.
 
 I was hoping to disable that and have nvim-orgmode take over so I can refile.

**Reply by captcapslock** [reply]
Hello there!

Give this a go:

1. Open your *.org document and check if you get "org" returned when executing this command:
`:lua print(vim.bo.filetype)`

2. Create an overwrite file for the plugin blink.cmp. depending on your neovim distro the location can vary, an example for astrovim is the following directory:
~/.config/nvim/lua/plugins

3. Here create a new file - for example "blink.lua" - and add these lines:
`return {
  "Saghen/blink.cmp",
  opts = {
    enabled = function() return not vim.tbl_contains({ "org", "txt" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt" end,
    },
}
`

4. Save the file with ":w" and reload the config files or restart neovim, if they dont auto-reload.

**Comment by esn89** [comment]
Thanks for your detailed write up, @captcapslock .  Unfortunately I am stuck at part 1 as I don't get "org" returned at all:

[image](https://imgur.com/a/ZYeMnZM)

**Reply by vicantwin** [reply]
Hey @esn89! 

I think the problem you have here is completely unrelated to OPs question. I believe your problem (according to your original message) is just **cmdline completion**, which is a completely different field in the opts table.

But I doubt you can use a function to change according to the current ft as I think the ft changes when you open a cmdline (I haven't tried it yet). So the only way to disable cmdline completions is just disabling it (as I've shown in the example).

```lua
{
  "saghen/blink.cmp",
  opts = {
    cmdline = {
      enabled = false,
    },
  },
}
```


Hope this helps!

**Comment by esn89** [comment]
Yeah.  I currently have that as false right now and nvim-orgmode works.

Ideally, I would like a way to have the best of both worlds:   cmdline completion, but NOT when nvim-orgmode is enabled.

**Reply by vicantwin** [reply]
Well this is beyond what blink.cmp can do right now, and the spec doesn't seem to accept a function either, unless @Saghen updates the plugin in the future.

**Reply by saghen** [reply]
If you want to disable cmdline conditionally, I'd use `sources.providers.cmdline.enabled = function() return your_logic end`

**Reply by esn89** [reply]
Anyone can hlep me out with the logic part of this?  What should my if statement be checking, in this case?
